src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -660,10 +668,151 @@ static inline void pio_clkdiv_restart_sm_mask(PIO pio, uint32_t mask) {
  */
 static inline void pio_enable_sm_mask_in_sync(PIO pio, uint32_t mask) {
     check_pio_param(pio);
+    check_sm_mask(mask);
     pio->ctrl |= ((mask << PIO_CTRL_CLKDIV_RESTART_LSB) & PIO_CTRL_CLKDIV_RESTART_BITS) |
                  ((mask << PIO_CTRL_SM_ENABLE_LSB) & PIO_CTRL_SM_ENABLE_BITS);
 }
 
+/*! \brief PIO interrupt source numbers for pio related IRQs
+ * \ingroup hardware_pio
+ */
+enum pio_interrupt_source {
+    pis_irq0 = PIO_INTR_SM0_LSB,
+    pis_irq1 = PIO_INTR_SM1_LSB,
+    pis_irq2 = PIO_INTR_SM2_LSB,
+    pis_irq3 = PIO_INTR_SM3_LSB,
```

Comment:
Looking at the names on the RHS of this enum, should these have been e.g. `pis_sm0` rather than `pis_irq0` ?

Reply:
I'm still trying to get to grips with the way that PIO-internal IRQs interact with system-level IRQs (it's not entirely clear in the datasheet), and I think maybe you're right; and in hindsight that fact that the `INTR` register labels these fields as `SM0 .. SM3` rather than `IRQ0 .. IRQ3` is a bit of a misnomer?

The description for `IRQ` says " ... There’s no fixed association between flags and state machines — any state machine can use any flag. ... The lower four of these flags are also routed out to system-level interrupt requests, alongside FIFO status interrupts — see e.g. IRQ0_INTE."

Reply:
yeah it is confusing, i named them without SM as they don't related to SMs in any particular way